pkg/plugin/repository/GlobalPluginRepository.go

```diff
@@ -881,11 +905,18 @@ func (impl *GlobalPluginRepositoryImpl) GetPluginParentMetadataByIds(ids []int)
 }
 
 func (impl *GlobalPluginRepositoryImpl) GetAllPluginMinData() ([]*PluginParentMetadata, error) {
+	return impl.GetAllPluginMinDataByType("")
+}
+
+func (impl *GlobalPluginRepositoryImpl) GetAllPluginMinDataByType(pluginType string) ([]*PluginParentMetadata, error) {
```

Comment:
`pluginType` shouldn't be just string , this method should accept the proper type.

Reply:
The method `GetAllPluginMinData()` is being called in multiple places internally. And I introduced `GetAllPluginMinDataByType(pluginType string)` recently for a list API. In the`GetAllPluginMinData()` we want all types of plugins. In order to reduce the impact area and testing scope, This implementation is being placed